extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -1676,4 +1700,88 @@ public void run()
       }
     };
   }
+
+  private Runnable computeAndEmitLag(final KafkaIndexTaskClient taskClient)
+  {
+    return new Runnable()
+    {
+      @Override
+      public void run()
+      {
+        try {
+          final Map<String, List<PartitionInfo>> topics = lagComputingConsumer.listTopics();
+          final List<PartitionInfo> partitionInfoList = topics.get(ioConfig.getTopic());
+          lagComputingConsumer.assign(Lists.transform(partitionInfoList, new Function<PartitionInfo, TopicPartition>()
+          {
+            @Override
+            public TopicPartition apply(PartitionInfo input)
+            {
+              return new TopicPartition(ioConfig.getTopic(), input.partition());
+            }
+          }));
+          final Map<Integer, Long> offsetsResponse = new HashMap<>();
```

Comment:
Actually may not be required as all the Future transform function will be called in the same thread.